io-sim/src/Control/Monad/IOSimPOR/Internal.hs

```diff
@@ -814,6 +816,11 @@ schedule thread@Thread{
       let thread' = thread { threadControl = ThreadControl k ctl }
       schedule thread' simstate
 
+    NewUnique k -> do
+      let thread'   = thread{ threadControl = ThreadControl (k nextUniq) ctl }
+          simstate' = simstate{ nextUniq = MkUnique (unMkUnique nextUniq + 1) }
+      schedule thread' simstate'
```

Comment:
If we implement `IOSimUnique` using `STRef s` (or `TVar m`) then the `MonadUnique` instance will require `MonadST` (or `MonadSTM`), which would be a disadvantage for users.  This approach doesn't require it :+1:, but it needs two implementations (`IOSim` and `IOSimPOR`).

Reply:
I don't really know the codebase well enough to weigh the merits of different paths; I just took the first one I found.